libhb/encavcodec.c

```diff
@@ -374,11 +455,36 @@ int encavcodecInit( hb_work_object_t * w, hb_job_t * job )
     {
         job->areBframes = 1;
     }
+
     if( ( job->mux & HB_MUX_MASK_MP4 ) && job->pass_id != HB_PASS_ENCODE_1ST )
     {
-        w->config->mpeg4.length = context->extradata_size;
-        memcpy( w->config->mpeg4.bytes, context->extradata,
-                context->extradata_size );
+        if (w->codec_param == AV_CODEC_ID_H264) // FIXME: h265 as well?
```

Comment:
is this VCE specific?

Reply:
It shouldn't be, both codecs have extra-data to copy into the stream.

Reply:
encavcodec.c isn't currently used for other h.264 encoders (or h.265 for that matter so you can remove the FIXME).  So this won't hurt.  I'm not sure if all other ffmpeg h.264 encoders would require this nalu parsing however.

Reply:
I'm not sure why this post-processing for the extra data is actually being done. Extra Data for the VCE encoders is simply more H264 Packets (full SPS/PPS blocks), so I'd love to know what this actually is for.